tests/sqlparser_postgres.rs

```diff
@@ -402,121 +402,201 @@ PHP	₱ USD $
 }
 
 #[test]
-fn test_copy_from() {
+fn parse_copy_from() {
     let stmt = pg().verified_stmt("COPY users FROM 'data.csv'");
     assert_eq!(
         stmt,
         Statement::Copy {
             table_name: ObjectName(vec!["users".into()]),
             columns: vec![],
-            filename: Some(Ident {
-                value: "data.csv".to_string(),
-                quote_style: Some('\'')
-            }),
+            to: false,
+            target: CopyTarget::File {
+                filename: "data.csv".to_string(),
+            },
+            options: vec![],
+            legacy_options: vec![],
+            values: vec![],
+        }
+    );
+
+    let sql = "COPY table (a, b) FROM 'file.csv' WITH 
+    (
+        FORMAT CSV,
+        FREEZE,
+        FREEZE TRUE,
+        FREEZE FALSE,
+        DELIMITER ',',
+        NULL '',
+        HEADER,
+        HEADER TRUE,
+        HEADER FALSE,
+        QUOTE '\"',
+        ESCAPE '\\',
+        FORCE_QUOTE (a, b),
+        FORCE_NOT_NULL (a),
+        FORCE_NULL (b),
+        ENCODING 'utf8'
+    )";
+    assert_eq!(
+        pg_and_generic().one_statement_parses_to(sql, ""),
+        Statement::Copy {
+            table_name: ObjectName(vec!["table".into()]),
+            columns: vec!["a".into(), "b".into()],
+            to: false,
+            target: CopyTarget::File {
+                filename: "file.csv".into()
+            },
+            options: vec![
+                CopyOption::Format("CSV".into()),
+                CopyOption::Freeze(true),
+                CopyOption::Freeze(true),
+                CopyOption::Freeze(false),
+                CopyOption::Delimiter(','),
+                CopyOption::Null("".into()),
+                CopyOption::Header(true),
+                CopyOption::Header(true),
+                CopyOption::Header(false),
+                CopyOption::Quote('"'),
+                CopyOption::Escape('\\'),
+                CopyOption::ForceQuote(vec!["a".into(), "b".into()]),
+                CopyOption::ForceNotNull(vec!["a".into()]),
+                CopyOption::ForceNull(vec!["b".into()]),
+                CopyOption::Encoding("utf8".into()),
+            ],
+            legacy_options: vec![],
             values: vec![],
-            delimiter: None,
-            csv_header: false,
-            to: false
         }
     );
+}
 
-    let stmt = pg().verified_stmt("COPY users FROM 'data.csv' DELIMITER ','");
+#[test]
```

Comment:
Would you mind keeping the existing test cases (to show how this PR changes their behavior) and the tests for the new behavior as new cases?

Reply:
restored in 13d0f3f